truth_set/java/structure/organisation/complex-code/ECommerce.java

```diff
@@ -0,0 +1,233 @@
+    private List<Discount> discounts;
+
+    public String getProductId() { return productId; }
+    public void setProductId(String id) { this.productId = id; }
+    public int getQuantity() { return quantity; }
+    public void setQuantity(int qty) { this.quantity = qty; }
+    public List<Option> getOptions() { return options; }
+    public void setOptions(List<Option> opts) { this.options = opts; }
+    public List<Discount> getDiscounts() { return discounts; }
+    public void setDiscounts(List<Discount> discs) { this.discounts = discs; }
+}
+
+class Shipping {
+    private String address;
+    private ShippingMethod method;
+    private double cost;
+
+    public String getAddress() { return address; }
+    public void setAddress(String addr) { this.address = addr; }
+    public ShippingMethod getMethod() { return method; }
+    public void setMethod(ShippingMethod m) { this.method = m; }
+    public double getCost() { return cost; }
+    public void setCost(double c) { this.cost = c; }
+}
+
+class Coupon {
+    private String code;
+    private boolean valid;
+    private String expiryDate;
+
+    public String getCode() { return code; }
+    public void setCode(String c) { this.code = c; }
+    public boolean isValid() { return valid; }
+    public void setValid(boolean v) { this.valid = v; }
+    public String getExpiryDate() { return expiryDate; }
+    public void setExpiryDate(String date) { this.expiryDate = date; }
+}
+
+class OrderInput {
+    private Customer customer;
+    private List<OrderItem> items;
+    private Shipping shipping;
+    private List<Coupon> coupons;
+
+    public Customer getCustomer() { return customer; }
+    public void setCustomer(Customer c) { this.customer = c; }
+    public List<OrderItem> getItems() { return items; }
+    public void setItems(List<OrderItem> i) { this.items = i; }
+    public Shipping getShipping() { return shipping; }
+    public void setShipping(Shipping s) { this.shipping = s; }
+    public List<Coupon> getCoupons() { return coupons; }
+    public void setCoupons(List<Coupon> c) { this.coupons = c; }
+}
+
+public class ECommerce {
+    public Map<String, Object> processOrder(OrderInput order) {
+        Map<String, Object> finalOrder = new HashMap<>();
+        if (!isValidOrder(order)) {
+            return null;
+        }
+
+        finalOrder.put("customerId", order.getCustomer().getId());
+        finalOrder.put("customerName", order.getCustomer().getName() != null ?
+                      order.getCustomer().getName() : "Valued Customer");
+
+        if (order.getCustomer().getLoyaltyStatus() != null &&
+            order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
+            finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
+        }
+
+        Map<String, Object> items = new HashMap<>();
+        finalOrder.put("items", items);
+        processOrderItems(order, items);
+        return finalOrder;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Method could be split for better organization</b></div>
<div id="fix">

The `processOrder` method could benefit from being split into smaller methods. Consider extracting customer processing and loyalty status handling into separate methods.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -111,19 +111,28 @@
  public Map<String, Object> processOrder(OrderInput order) {
      Map<String, Object> finalOrder = new HashMap<>();
      if (!isValidOrder(order)) {
          return null;
      }
 -    finalOrder.put("customerId", order.getCustomer().getId());
 -    finalOrder.put("customerName", order.getCustomer().getName() != null ?
 -                  order.getCustomer().getName() : "Valued Customer");
 -
 -    if (order.getCustomer().getLoyaltyStatus() != null &&
 -        order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
 -        finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
 -    }
 +    processCustomerDetails(order, finalOrder);
 +    processLoyaltyStatus(order, finalOrder);
      Map<String, Object> items = new HashMap<>();
      finalOrder.put("items", items);
      processOrderItems(order, items);
      return finalOrder;
 +}
 +
 +private void processCustomerDetails(OrderInput order, Map<String, Object> finalOrder) {
 +    finalOrder.put("customerId", order.getCustomer().getId());
 +    finalOrder.put("customerName", order.getCustomer().getName() != null ?
 +                  order.getCustomer().getName() : "Valued Customer");
 +}
 +
 +private void processLoyaltyStatus(OrderInput order, Map<String, Object> finalOrder) {
 +    if (order.getCustomer().getLoyaltyStatus() != null &&
 +        order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
 +        finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
 +    }
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/8#issuecomment-2561208575>#e3503f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged